### PR TITLE
Fix reusable block crash when converting a just created reusable block to blocks.

### LINKED ISSUE #26421: Reusable block can't be converted to a regular block
I have 2 different reusable blocks which worked fine in the past.
When I add them now, I can't convert them to a regular block. 

I have Gutenberg 9.2.1 installed.
Rolled back to 9.2, the same issue.
Rolled back to 9.1.1, everything is working again.

So starting with 9.2. there is something breaking the "convert to a regular block" function to make the block editable.

### PR DESCRIPTION
**Pull request by talldan**
## Description
Fixes #26421

Reusable blocks crash on conversion when they have unsaved edits.

This happens because the entity `content` prop is passed to the blocks `parse` function. Usually that's ok, because it'll be a string. But when a reusable block has unsaved edits, `content` is a function that `parse` doesn't know how to handle.

The content function is created here:
https://github.com/WordPress/gutenberg/blob/60314e8e879579058164e30514cf217a47a00dd5/packages/core-data/src/entity-provider.js#L183-L187

It looks like this function is called on `saveEntityRecord`, but is otherwise stored internally as a function:
https://github.com/WordPress/gutenberg/blob/5d3681a8af68c50da6921902284276838657f1ea/packages/core-data/src/actions.js#L386-L394

This became an issue in #25859, where reusable blocks were switched over to using the entity system. Before that they did a plain `apiFetch` and as far as I can tell always dealt with string `content`.

## How has this been tested?
1. Add a paragraph block and type the text '1' within it.
2. Convert the paragraph to reusable
3. Edit the reusable block, change the paragraph text to '12'.
4. Try converting the reusable block back to regular blocks.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by talldan** [comment]
It seems this has been a problem since #25859 switched from using `apiFetch` to using core-data entities.

**Comment by talldan** [comment]
I discovered this is actually really easy to reproduce by editing a created reusable block and trying to convert it back to regular blocks before saving. That made the test case much easier to write.

I think this problem will have been made more common by recent changes to reusable blocks. Back when they had a more distinct editing/saved state it would be difficult to do this. The bug reports indicate it probably happened only with particular blocks, but now it could quite easily happen with any block.

File: packages/e2e-tests/specs/editor/various/reusable-blocks.test.js [hunk 1]
```diff
@@ -297,4 +297,29 @@ describe( 'Reusable blocks', () => {
 			expect( content ).toEqual( 'Awesome Paragraph modified' );
 		} );
 	} );
+
+	// Check for regressions of https://github.com/WordPress/gutenberg/issues/26421.
+	it( 'allows conversion back to blocks when the reusable block has unsaved edits', async () => {
```
Comment (youknowriad) [inline packages/e2e-tests/specs/editor/various/reusable-blocks.test.js:302]: Thanks for the test.

**Review by youknowriad — APPROVED** [review]
Thanks for the fix.

I think ideally these "function as edit" behavior should be something only internal to core-data but a change like that would require a lot more thinking and impact so this is a good fix for now.

**Pull request merged**